wgsl/index.bs

```diff
@@ -3538,7 +4033,61 @@ compound_statement
   : BRACE_LEFT statements BRACE_RIGHT
 </pre>
 
-## Assignment TODO ## {#assignment}
+## Assignment ## {#assignment}
+
+An <dfn noexport>assignment statement</dfn> replaces the contents of a variable,
```

Comment:
Do we want to talk about memory here? It seems strange to introduce the concept of memory locations and such, only to completely skip it here where it's actually being written.

Reply:
The semantics given in the table tells us:

> Evaluates _e_, evaluates _r_, then **replaces the contents of memory referenced by _r_** with the value computed for _e_.

(bold added)

A memory reference corresponds to a memory view, and a memory view _is a set of memory locations_ (along with an interpretation type).

I thought that was sufficiently spelled out.

Reply:
Maybe you want this to also say.  "Writes to the memory locations associated with |r|."?

Reply:
Or to write

> Evaluates _e_, evaluates _r_, then replaces the contents of memory locations referenced by _r_ with the value computed for _e_.

i.e. insert "locations" there?

Reply:
Maybe this is better:

> Evaluates e, evaluates r, then writes the value computed for e into the memory locations referenced by r.

I'll go with that.

Reply:
thanks!